openbas-api/src/main/java/io/openbas/service/ChallengeService.java

```diff
@@ -113,7 +119,16 @@ public ChallengesReader playerChallenges(String exerciseId, User user) {
                 injectExpectation -> {
                   Challenge challenge = injectExpectation.getChallenge();
                   challenge.setVirtualPublication(injectExpectation.getCreatedAt());
-                  return new ChallengeInformation(challenge, injectExpectation);
+                  InjectStatus injectStatus =
+                      injectExpectation.getInject().getStatus().orElseThrow();
```

Comment:
Even if this should never happen, could you add a descriptive exception to the orElseThrow?

Reply:
Good point